CRM/Utils/SQL/Select.php

```diff
@@ -323,13 +323,17 @@ public function having($exprs, $args = NULL) {
    *
    * @param string|array $exprs list of SQL expressions
    * @param null|array $args use NULL to disable interpolation; use an array of variables to enable
-   * @return CRM_Utils_SQL_Select
+   * @param int $weight
+   * @return \CRM_Utils_SQL_Select
    */
-  public function orderBy($exprs, $args = NULL) {
+  public function orderBy($exprs, $args = NULL, $weight = NULL) {
     $exprs = (array) $exprs;
+    if ($weight === NULL) {
+      $weight = count($this->orderBys);
```

Comment:
@totten So you are trying to solve the problem of mixed input (some orderBys have a weight and others do not)? Note that this will be a very rare occurrence.